### DISCUSSION TITLE
Thinking of in the near future making a pure Rust version of fltk-rs

### DISCUSSION THREAD

**Discussion by Raj2032** [fltk-rs/fltk-rs#1195] [Ideas]
fltk-rs seems like the most simpliest library to use. It doesn't lock you into some kind of framework, it has retained graphics capabilities, it is easy to use and understand. But there are some drawbacks, for example at the level of fltk, they only support opengl, and it is basically using C lang and fltk-rs is just a binding.

I was thinking if I play around with this to basically work on re-writing everything from scratch but using the same API.

Is this something anyone would be interested in?

**Comment by MoAlyousef** [comment]
Writing a GUI library from scratch is a major undertaking. I’m working on a wasm-based frontend library, [livid](https://github.com/MoAlyousef/livid-rs), which mimics the fltk-rs api. It’s easier than writing a GUI from scratch, but it requires webtech. That said, an advantage of webtech is that it can target web and desktop (via webview) (and potentially mobile).

**Reply by Raj2032** [reply]
Interesting I will have a look at it thanks :)
